File: Java/src/com/dbc/algorithm/ReferOffer/HeWeiSdeLianXuZhengShuXuLieLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class HeWeiSdeLianXuZhengShuXuLieLcof {
    public int[][] findContinuousSequence(int target) {
        Queue<Integer> queue = new LinkedList<>();
        List<int[]> res = new ArrayList<>();
        int total = 0;
        for (int i = 1; i <= target; i++) {
            while (total > target) {
                total -= queue.poll();
            }
            if (total == target) {
                int[] temp = new int[i - queue.peek()];
                for (int j = queue.peek(); j < i; j++)
                    temp[j - queue.peek()] = j;
                res.add(temp);
            }
            queue.offer(i);
            total += i;
        }
        return res.toArray(new int[res.size()][]);
    }
}

File: src/tmp/Python/refer-offer/he_wei_sde_lian_xu_zheng_shu_xu_lie_lcof.py
from typing import *

def findContinuousSequence(target: int) -> List[List[int]]:
    total, queue, res = 0, Deque(), list()
    for i in range(1, target + 1):
        if total > target:
            while total > target:
                total -= queue.popleft()
        if total == target:
            res.append(list(queue.copy()))
        
        queue.append(i)
        total += i
    
    return res

if __name__ == "__main__":
    print(findContinuousSequence(9))
